route/build/src/tracer.f90

```diff
@@ -0,0 +1,364 @@
+     enddo
+   end if
+   write(iulog,'(A,1X,G15.4)') ' RCHFLX_out(iEns,segIndex)%basin_solute=',RCHFLX_out(iEns,segIndex)%basin_solute
+ endif
+
+ ! compute mass flux out and average concentration
+ call comp_mass_flux(idxRoute,                   &  ! input: routing method index
+                     RPARAM_in(segIndex),        &  ! input: parameter at segIndex reach
+                     Cupstream,                  &  ! input: total discharge at top of the reach being processed
+                     Clat,                       &  ! input: lateral mass flow [mg/s]
+                     isHW,                       &  ! input: is this headwater basin?
+                     subrch_state,               &  ! inout:
+                     RCHFLX_out(iens,segIndex),  &  ! inout: updated fluxes at reach
+                     verbose,                    &  ! input: reach index to be examined
+                     ierr, cmessage)                ! output: error control
+ if(ierr/=0)then
+    write(message, '(A,1X,I12,1X,A)') trim(message)//'/segment=', NETOPO_in(segIndex)%REACHID, '/'//trim(cmessage)
+    return
+ endif
+
+ if(verbose)then
+   write(iulog,'(A,1X,G15.4)') ' RCHFLX_out(iens,segIndex)%reach_solute_flux=', RCHFLX_out(iens,segIndex)%ROUTE(idxRoute)%reach_solute_flux
+ endif
+
+ if (RCHFLX_out(iens,segIndex)%ROUTE(idxRoute)%reach_solute_flux < 0) then
+   write(iulog,'(A,1X,G12.5,1X,A,1X,I9)') ' ---- NEGATIVE MASS = ', RCHFLX_out(iens,segIndex)%ROUTE(idxRoute)%reach_solute_flux, &
+         'at ', NETOPO_in(segIndex)%REACHID
+ end if
+
+ call comp_reach_mb(NETOPO_in(segIndex)%REACHID, idxDW, Cupstream, Clat, RCHFLX_out(iens,segIndex), verbose, lakeFlag=.false.)
+
+ END SUBROUTINE constituent_rch
+
+ ! *********************************************************************
+ ! subroutine: solve diffuisve wave equation
+ ! *********************************************************************
+ SUBROUTINE comp_mass_flux(idxRoute,       & ! input: routing method index
+                           rch_param,      & ! input: river parameter data structure
+                           Cupstream,      & ! input: discharge from upstream
+                           Clat,           & ! input: lateral mass flux into chaneel [mg/s]
+                           isHW,           & ! input: is this headwater basin?
+                           subrch_state,   & ! inout: state at a sub-reach
+                           rflux,          & ! inout: reach flux at a reach
+                           verbose,        & ! input: reach index to be examined
+                           ierr,message)
+ ! ----------------------------------------------------------------------------------------
+ ! Description
+ !
+ ! ----------------------------------------------------------------------------------------
+ USE globalData, ONLY : nMolecule   ! number of internal nodes for finite difference (including upstream and downstream boundaries)
```

Comment:
I'm unclear as to the meaning here. Molecule sounds to me like it's referring to a molecule in terms of chemistry. So H2O or CO2 or something. But, it sounds like this an integer referring to number of reaches or something?

Reply:
Yes, nMolecule is a number of sub-reach points for finite difference, and used for water routing too. For a tracer, nMolecule is confusing, so may need to renamed to something else. Here, nMolecule is not used and so removed.